test_matrix.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+from Matrix import Matrix
+
+
+class Test_matrix(unittest.TestCase):
+    def setUp(self):
+        self.x = Matrix(((1, 2), (3, 4)))
+        self.y = Matrix(((4, 3), (2, 1)))
+        self.zeros = Matrix(((0, 0), (0, 0)))
+        self.tens = Matrix(((10, 10), (10, 10)))
+        self.negative = Matrix(((-1, -1), (-1, -1)))
+        self.three = Matrix(((1, 2, 0), (0, 1, 2), (2, 0, 1)))
+        self.half = Matrix(((0.5, 0.5), (0.5, 0.5)))
+
+    def test_init_failed(self):
+        with self.assertRaises(ValueError):
+            (Matrix("hello"))
+        with self.assertRaises(ValueError):
+            (Matrix(((2, 1), (1, 2, 3))))
+
+    def test_add(self):
+        self.assertEqual(self.x + self.y, Matrix(((5, 5), (5, 5))))
+        self.assertEqual(self.x + Matrix(((-2, -3), (-4, -5))), Matrix(((-1, -1), (-1, -1))))
+        self.assertEqual(self.three + self.three, Matrix(((2, 4, 0), (0, 2, 4), (4, 0, 2))))
```

Comment:
This is not good, you use __eq__ to compare between the function you wrote to the result.
There might be bug in adding matrices, but since there is also bug in __eq__ the tests might miss it
(This is correct for all tests)

Reply:
?